File: src/Brainbrakers.Mobile/app/src/main/java/com/keyldev/brakerspodcast/Models/Podcast.java
package com.keyldev.brakerspodcast.Models;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class Podcast {
    public UUID id;
    public String title = "";
    public String description = "";
    public String releaseDate;
    public String coverImageURL = "";
    public String rssFeedURL = "";// ?? не факт что понадобится
    public List<Object> authors = new ArrayList<>();
    //public virtual ICollection<User> Authors  = new List<User>();
    public List<Object> subscribers = new ArrayList<Object>();
    public List<Object> categories = new ArrayList<Object>();
    public List<Object> keywords = new ArrayList<Object>();
    public List<Episode> episodes = new ArrayList<Episode>();

    public double rating = 0.0;
    public int duration = 0;
    public String language = "en";
    public boolean isAdultContent = false;
    public boolean isUserMaker = true;
}

File: src/Brainbrakers.API.Models/Models/Podcast.cs

namespace podcast_api.Models
{
    public class Podcast
    {
        public Guid Id { get; set; }
        public string Title { get; set; } = "";
        public string Description { get; set; } = "";
        public DateTime ReleaseDate { get; set; }
        public string CoverImageURL { get; set; } = "";
        public string RssFeedURL { get; set; } = "";// 
        public virtual ICollection<PodcastAuthors> Authors { get; set; } = new List<PodcastAuthors>();
        
        public virtual ICollection<Subscription> Subscribers { get; set; } = new List<Subscription>();
        public virtual ICollection<Categories> Categories { get; set; } = new List<Categories>();
        public virtual ICollection<Keywords> Keywords { get; set; } = new List<Keywords>();
        public virtual ICollection<Episode> Episodes { get; set; } = new List<Episode>();

        public double Rating { get; set; } = 0.0;
        public int Duration { get; set; } = 0;
        public string Language { get; set; } = "en";
        public bool IsAdultContent { get; set; } = false;

    }
}
